uqcsbot/utils/uq_course_utils.py

```diff
@@ -105,6 +106,72 @@ def _estimate_current_semester() -> SemesterType:
             return "Summer"
 
 
+@dataclass
+class AssessmentItem:
+    course_name: str
+    task: str
+    due_date: str
+    weight: str
+
+    def get_parsed_due_date(self):
+        """
+        Returns the parsed due date for the given assessment item as a datetime
+        object. If the date cannot be parsed, a DateSyntaxException is raised.
+        """
+        if self.due_date == "Examination Period":
+            return get_current_exam_period()
+        parser_info = parser.parserinfo(dayfirst=True)
+        try:
+            # If a date range is detected, attempt to split into start and end
+            # dates. Else, attempt to just parse the whole thing.
+            if " - " in self.due_date:
+                start_date, end_date = self.due_date.split(" - ", 1)
+                start_datetime = parser.parse(start_date, parser_info)
+                end_datetime = parser.parse(end_date, parser_info)
+                return start_datetime, end_datetime
+            due_datetime = parser.parse(self.due_date, parser_info)
+            return due_datetime, due_datetime
+        except Exception:
+            raise DateSyntaxException(self.due_date, self.course_name)
+
+    def is_after(self, cutoff: datetime):
+        """
+        Returns whether the assessment occurs after the given cutoff.
+        """
+        try:
+            start_datetime, end_datetime = self.get_parsed_due_date()
+        except DateSyntaxException:
+            # TODO bot.logger.error(e.message)
+            # If we can't parse a date, we're better off keeping it just in case.
+            # TODO(mitch): Keep track of these instances to attempt to accurately
```

Comment:
Noting that these TODO's are 2 years old. At a glance, I can't tell how simple these are to fix - it looks like it's failing 'safely', so maybe just delete the TODO's?

Reply:
I agree with removing them. I tried to find cases where it failed to parse, and where it did, it was because there was no good date to use.